quetz/tasks/mirror.py

```diff
@@ -235,38 +261,24 @@ def handle_repodata_package(
         total_size += size
         file.file.seek(0)
 
-    dao.assert_size_limits(channel_name, total_size)
+    # create package in database
+    # channel_data = _load_remote_channel_data(remote_repository)
+    # create_packages_from_channeldata(channel_name, user_id, channel_data, dao)
 
-    @retry(
-        stop=stop_after_attempt(3),
-        wait=wait_exponential(multiplier=1, min=4, max=10),
-        after=after_log(logger, logging.WARNING),
-    )
-    def _upload_package(file, channel_name, subdir):
-        dest = os.path.join(subdir, file.filename)
-
-        try:
-            file.file.seek(0)
-            logger.debug(
-                f"uploading file {dest} from channel {channel_name} to package store"
-            )
-            pkgstore.add_package(file.file, channel_name, dest)
-
-        except AttributeError as e:
-            logger.error(f"Could not upload {file}, {file.filename}. {str(e)}")
-            raise TryAgain
+    # validate quota
+    dao.assert_size_limits(channel_name, total_size)
 
     pkgstore.create_channel(channel_name)
-    nthreads = config.general_package_unpack_threads
 
     with TicToc("upload file without extracting"):
+        nthreads = config.general_package_unpack_threads
         with ThreadPoolExecutor(max_workers=nthreads) as executor:
             for file, package_name, metadata in files_metadata:
                 if proxylist and package_name in proxylist:
                     # skip packages that should only ever be proxied
                     continue
                 subdir = get_subdir_compat(metadata)
-                executor.submit(_upload_package, file, channel_name, subdir)
+                executor.submit(_upload_package, file, channel_name, subdir, pkgstore)
```

Comment:
Is there a behavior change here? Looks like you just moved the function def out of the current closure into the file scope. Anything else?

Reply:
Nope, that should be it. And because of that, I had to pass the `pkgstore` as well.